connector/forwardconnector/forward.go

```diff
@@ -0,0 +1,134 @@
+)
+
+const (
+	typeStr = "forward"
+)
+
+// NewFactory returns a connector.Factory.
+func NewFactory() connector.Factory {
+	return connector.NewFactory(
+		typeStr,
+		createDefaultConfig,
+		connector.WithTracesToTraces(createTracesToTraces, component.StabilityLevelDevelopment),
+		connector.WithMetricsToMetrics(createMetricsToMetrics, component.StabilityLevelDevelopment),
+		connector.WithLogsToLogs(createLogsToLogs, component.StabilityLevelDevelopment),
+	)
+}
+
+// createDefaultConfig creates the default configuration.
+func createDefaultConfig() component.Config {
+	return &struct{}{}
+}
+
+// createTracesToTraces creates a trace receiver based on provided config.
+func createTracesToTraces(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Traces,
+) (connector.Traces, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Traces = nextConsumer
+	return conn, nil
+}
+
+// createMetricsToMetrics creates a metrics receiver based on provided config.
+func createMetricsToMetrics(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Metrics,
+) (connector.Metrics, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Metrics = nextConsumer
+	return conn, nil
+}
+
+// createLogsToLogs creates a log receiver based on provided config.
+func createLogsToLogs(
+	_ context.Context,
+	set connector.CreateSettings,
+	cfg component.Config,
+	nextConsumer consumer.Logs,
+) (connector.Logs, error) {
+	comp := connectors.GetOrAdd(cfg, func() component.Component {
+		return &forwardConnector{}
+	})
+
+	conn := comp.Unwrap().(*forwardConnector)
+	conn.Logs = nextConsumer
+	return conn, nil
+}
+
+// This is the map of already created forward connectors for particular configurations.
+// We maintain this map because the Factory is asked trace, metric, and log receivers
+// separately but they must not create separate objects. When the connector is shutdown
+// it should be removed from this map so the same configuration can be recreated successfully.
+var connectors = sharedcomponent.NewSharedComponents()
+
+// forwardConnector is used to pass signals directly from one pipeline to another.
+// This is useful when there is a need to replicate data and process it in more
+// than one way. It can also be used to join pipelines together.
+type forwardConnector struct {
+	consumer.Traces
+	consumer.Metrics
+	consumer.Logs
+}
+
+func (c *forwardConnector) Capabilities() consumer.Capabilities {
+	return consumer.Capabilities{MutatesData: false}
+}
+
+func (c *forwardConnector) Start(_ context.Context, host component.Host) error {
+	return nil
+}
+
+func (c *forwardConnector) Shutdown(ctx context.Context) error {
+	return nil
+}
```

Comment:
Include the `component.StartFunc` and `component.ShutdownFunc` instead.

Reply:
Done